From: Aurelien Jarno <aurelien@aurel32.net>
Date: Sat, 25 Apr 2015 14:16:16 +0200
Subject: [PATCH] ppc64: fix struct stat
Bug-Debian: https://bugs.debian.org/783292
Forwarded: http://www.zytor.com/pipermail/klibc/2016-January/003881.html

On ppc64 the struct stat defined by klibc matches the kernel one.
However it contains implicit padding before the st_rdev field due to the
64-bit alignement. For internal reasons, klibc defines st_rdev as a pair
of 32-bit values (using the __stdev64 macro). They only need to be
32-bit aligned and as a consequence st->st_rdev is incorrectly defined.

The solution is to add an explicit padding in the structure. This fixes
the resume binary on ppc64 BE and LE, and probably other things.

Signed-off-by: Aurelien Jarno <aurelien@aurel32.net>
---
 usr/include/arch/ppc64/klibc/archstat.h | 1 +
 1 file changed, 1 insertion(+)

--- a/usr/include/arch/ppc64/klibc/archstat.h
+++ b/usr/include/arch/ppc64/klibc/archstat.h
@@ -12,6 +12,7 @@ struct stat {
 	mode_t		st_mode;
 	uid_t 		st_uid;
 	gid_t 		st_gid;
+	unsigned int	__pad1;
 	__stdev64	(st_rdev);
 	off_t		st_size;
 	unsigned long  	st_blksize;
